components/dashboard/src/feedback-form/FeedbackComponent.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useState } from "react";
+import starry from "../images/feedback/starry-emoji.svg";
+import happy from "../images/feedback/happy-emoji.svg";
+import meh from "../images/feedback/meh-emoji.svg";
+import crying from "../images/feedback/crying-emoji.svg";
+import { trackEvent } from "../Analytics";
+
+function FeedbackComponent(props: { onClose: () => void; onSubmit: () => void; isModal: boolean }) {
+    const [text, setText] = useState<string>("");
+    const [selectedEmoji, setSelectedEmoji] = useState<number | undefined>();
+
+    const height = props.isModal ? "300px" : "";
+
+    const onSubmit = () => {
+        if (selectedEmoji) {
+            const feedbackObj = {
+                score: selectedEmoji,
+                feedback: text,
+                href: window.location.href,
+                path: window.location.pathname,
+            };
+            trackEvent("feedback_submitted", feedbackObj);
+        }
+
+        props.onSubmit();
+    };
+
+    const handleClick = (emojiScore: number) => {
+        setSelectedEmoji(emojiScore);
+    };
+
+    const emojiGroup = (width: number) => {
```

Comment:
I think this can become an `EmojiGroup` component, taking `width` as a prop, rather than a function.

Reply:
not necessary, but I always find it odd when a function returns a react element but isn't a component. It doesn't need to be moved to a separate file.

✅ and up to you whether you change it or not 😄

Reply:
Ah, interesting thought! I do think in the team there are different views on what best practices we can do in React, and maybe worth having a discussion on 🤩 I will skip this extraction for this one, but keeping my mind open 🤗